src/activities/handlers.clj

```diff
@@ -49,71 +49,82 @@
                 str)})))
 
 ;; GET /activity/new
-(defn new-activity-form [req]
+(defn new-activity-form [{:keys [reitit.core/router] :as req}]
   (let [user-uuid (user/req->uuid req)
-        db (crux/db (:crux req))]
+        db        (crux/db (:crux req))]
     (if (crux/entity db user-uuid)
       (response req {:title "New Activity"} (views/activity-form req))
-      {:status 302
-       :headers {"Location" (path req :activities.system/login-form)}})))
+      {:status  302
+       :headers {"Location" (path router :activities.system/login-form)}})))
 
 ;; POST /activity
-(defn create-activity [req]
-  (let [node        (:crux req)
-        activity    (activity/req->new-activity req)
-        activity-id (str (:crux.db/id activity))]
-    (submit-tx node [[:crux.tx/put activity]])
-    {:status 303
-     :headers {"Location" (path req :activities.system/activity {:id activity-id})}}))
+(defn create-activity
+  [{:keys [crux params session reitit.core/router]}]
+  (let [new-uuid   (UUID/randomUUID)
+        creator    (:identity session)
+        attributes (assoc params :uuid new-uuid :creator creator)
+        activity   (activity/new-activity attributes)
+        path       (path router :activities.system/activity {:id new-uuid})]
+    (if (s/valid? :activities/activity activity)
+      (do (submit-tx crux [[:crux.tx/put activity]])
+          {:status  303
+           :headers {"Location" path}})
+      (s/explain :activities/activity activity)))) ;should it throw?
```

Comment:
you probably should return a normal response. Something like `{:status 400 :body (s/explain ... )}` (look up the most appropriate http code)

Reply:
I have realized that the function had that error page but I removed it trying to get it into a working state a few commits ago and forgot to add it again. Thanks.